### PR TITLE
JRF mii Domain test class/infra for the mii RCU functionality testing

### PR DESCRIPTION
**Pull request by maggiehe00**
The latest clean Kind new Jenkins run after addressing Vanaja's comment
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2996/testReport/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+        replicaCount,
+        jrfMiiImage);
+
+    // create model in image domain
+    createDomainAndVerify(domain, jrfDomainNamespace);
+
+    //check admin service created
+    logger.info("Check admin service {0} is created in namespace {1}",
+        adminServerPodName, jrfDomainNamespace);
+    checkServiceExists(adminServerPodName, jrfDomainNamespace);
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java:215]: combine checkServiceExists() and checkPodReady()

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+    // check admin server pod is ready
+    logger.info("Wait for admin server pod {0} to be ready in namespace {1}",
+        adminServerPodName, jrfDomainNamespace);
+    checkPodReady(adminServerPodName, domainUid, jrfDomainNamespace);
+
+    // check managed server services created
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Check managed service {0} is created in namespace {1}",
+          managedServerPrefix + i + "-c1", jrfDomainNamespace);
+      checkServiceExists(managedServerPrefix + i + "-c1", jrfDomainNamespace);
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java:226]: combine checkServiceExists() and checkPodReady()

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+        + "dbUrl %s", RCUSCHEMAPREFIX, dbNamespace, dbUrl));
+
+    // install operator and verify its running in ready state
+    installAndVerifyOperator(opNamespace, jrfDomainNamespace);
+
+    logger.info("For ItJrfMiiDomain using DB image: {0}, FMW image {1}",
+        DB_IMAGE_TO_USE_IN_SPEC, FMWINFRA_IMAGE_TO_USE_IN_SPEC);
+
+  }
+
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java:129]: Need javadoc.  Step-by-step information for  the usecase.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+          managedServerPrefix + i + "-c1", jrfDomainNamespace);
+      checkServiceExists(managedServerPrefix + i + "-c1", jrfDomainNamespace);
+    }
+
+    // check managed server pods are ready
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Wait for managed pod {0} to be ready in namespace {1}",
+          managedServerPrefix + i + "-c1", jrfDomainNamespace);
+      checkPodReady(managedServerPrefix + i + "-c1", domainUid, jrfDomainNamespace);
+    }
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java:234]: Verify few JRF scenarios.  Check accessibility of default JRF application or check em console.

**Review by anpanigr — APPROVED** [review]
LGTM

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+    installAndVerifyOperator(opNamespace, jrfDomainNamespace);
+
+    logger.info("For ItJrfMiiDomain using DB image: {0}, FMW image {1}",
+        DB_IMAGE_TO_USE_IN_SPEC, FMWINFRA_IMAGE_TO_USE_IN_SPEC);
+
+  }
+
+  /**
+   * Create a basic JRF model in image domain.
+   * Verify Pod is ready and service Exists for both admin server and managed servers.
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java:137]:
```suggestion
   * Verify Pod is ready and service exists for both admin server and managed servers.
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+  private static final String RCUSYSPASSWORD = "Oradoc_db1";
+  private static final String RCUSCHEMAUSERNAME = "myrcuuser";
+  private static final String RCUSCHEMAPASSWORD = "Oradoc_db1";
+  private static final String modelFile = "model-singleclusterdomain-sampleapp-jrf.yaml";
+
+  private static String dbUrl = null;
+  private static LoggingFacade logger = null;
+
+  private final String domainUid = "jrfdomain-mii";
+  //private final String wlSecretName = domainUid + "-weblogic-credentials";
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java:84]: remove commented code

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+  //private final String rcuSecretName = domainUid + "-rcu-credentials";
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java:85]: remove commented code

**Comment by maggiehe00** [comment]
Pani's review comments have been addressed. Jenkins job posted above reflected the latest code change.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+    int nodePort = getServiceNodePort(
+           jrfDomainNamespace, getExternalServicePodName(adminServerPodName), "default");
+    assertTrue(nodePort != -1,
+          "Could not get the default external service node port");
+    logger.info("Found the default service nodePort {0}", nodePort);
+    String curlCmd1 = "curl -s -L --show-error --noproxy '*' "
+        + " http://" + K8S_NODEPORT_HOST + ":" + nodePort
+        + "/em --write-out %{http_code} -o /dev/null";
+    logger.info("Executing default nodeport curl command {0}", curlCmd1);
+    assertTrue(callWebAppAndWaitTillReady(curlCmd1, 5));
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java:240]: add a message when it fails

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java [hunk 9]
```diff
@@ -0,0 +1,327 @@
...
+                            .model(new Model()
+                                    .domainType("JRF")
+                                    .runtimeEncryptionSecret(encryptionSecretName))
+                            .addSecretsItem(rcuAccessSecretName)
+                            .introspectorJobActiveDeadlineSeconds(600L)));
+
+    return domain;
+  }
+
+  private void checkAppIsRunning(
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java:299]: doesn't look like you are calling this method, delete it

**Review by vanajamukkara — CHANGES_REQUESTED** [review]
looks good, left some minor comments

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java (hunk 9 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java:299]: Yeah, for this JIRA's work I don't need it. Let me remove it for now.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java (hunk 5 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java]: Done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java (hunk 6 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java]: Done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java (hunk 8 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java]: Done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java (hunk 9 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItJrfMiiDomain.java]: Done

**Review by vanajamukkara — APPROVED** [review]
LG

**Pull request merged**